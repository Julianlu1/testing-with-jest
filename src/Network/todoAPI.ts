export async function getAllTodos(){
    const result = await fetch('https://jsonplaceholder.typicode.com/todos?&_limit=5')
    const data = await result.json().then(data =>{
        return data;
    });
    return data;
}


