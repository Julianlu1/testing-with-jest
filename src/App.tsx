import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import ToDoList from './components/ToDoList';
import AddToDo from './components/AddToDo';

import Slider, { Range } from 'rc-slider';
import 'rc-slider/assets/index.css';

import {getAllTodos} from './Network/todoAPI';
import { title } from 'process';

interface State {
  items: Array<Item>;
  value : number;
  deleteItemParent: Function;

}

interface Item {
  title: string;
  completed: boolean;
}

class App extends Component {
    
  public deleteItemParent = (index : any) =>{
    let array = [...this.state.items]; // make a separate copy of the array
    array.splice(index, 1); // Remove item by index
    this.setState({items: array});
  }
  public state : State = {
    items: [],
    value : 5,
    deleteItemParent : this.deleteItemParent
  };


  public addToDo = (state: Item) =>{
    let array = [...this.state.items]; // make a separate copy of the array
      array.push(state)
      this.setState({
        items: array
      });
  }

  getAllTodos = async () => {
    await fetch(
      `https://jsonplaceholder.typicode.com/todos?&_limit=${this.state.value}`
    )
      .then((response) => response.json())
      .then((json) => {
        this.setState({ items: json })
      });
  };

 

  componentDidMount() {
    this.getAllTodos();
  }

 onSliderChange = (value : any) => {
  this.setState({
    value,
  });
};

componentDidUpdate(prevProps : any, prevState: any) {
  // Updates todo's if state changed
  if (prevState.value !== this.state.value) {
    this.getAllTodos();
    console.log(this.state.items)
  }
}

testAction = () =>{
  this.setState({
    value : 80
  });
}

  render(){
    return (
      <div className="App">
        <button onClick={this.testAction} id="testbtn">Test</button>
            <AddToDo addToDoParent={this.addToDo}/>
            <div>
                Todo's : {this.state.value}
            </div>
            <Slider
            className="slider"
            style={{width: "25%", margin: "0 auto"}}
          value={this.state.value}
          onChange={this.onSliderChange}
        />
            <ToDoList items={this.state.items} deleteFromParent={this.deleteItemParent}/>
      </div>
    );
  }
 
}

export default App;
