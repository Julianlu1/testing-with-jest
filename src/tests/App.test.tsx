import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
import {getAllTodos} from '../Network/todoAPI';


test('renders the component', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});

it('should retrieve data from the api endpoint', async () =>{
  const todos : Array<any> = await getAllTodos();
  expect(todos).toHaveLength(5);
})

it("has 5 items in the state", async () => {
  const wrapperApp : any = shallow(<App />);
  
  await new Promise(res => setTimeout(res,20));
  const items : Array<any> = wrapperApp.state('items');

  expect(items).toHaveLength(5);
});


