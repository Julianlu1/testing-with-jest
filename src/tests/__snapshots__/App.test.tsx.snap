// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="App"
>
  <button
    id="testbtn"
    onClick={[Function]}
  >
    Test
  </button>
  <AddToDo
    addToDoParent={[Function]}
  />
  <div>
    Todo's : 
    5
  </div>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className="slider"
    disabled={false}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Array [
        Object {},
      ]
    }
    included={true}
    marks={Object {}}
    max={100}
    min={0}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={Object {}}
    reverse={false}
    step={1}
    style={
      Object {
        "margin": "0 auto",
        "width": "25%",
      }
    }
    trackStyle={
      Array [
        Object {},
      ]
    }
    value={5}
    vertical={false}
  />
  <ToDoList
    deleteFromParent={[Function]}
    items={Array []}
  />
</div>
`;
